File: StriverA2Z/Java/Arrays/Easy/LargestElementInArray.java
// O(n) -TC

public class LargestElementInArray {
    public static void main(String[] args) {
        int[] arr = {3, 1, 2, 5, 4};
        largestElement(arr);
    }

    static void largestElement(int[] arr) {
        int largest = arr[0];
        for (int j : arr) {
            if (largest < j) {
                largest = j;
            }
        }
        System.out.println("The largest element in the array is: " + largest);
    }
}

File: StriverA2Z/C++/3_ArraysQuestion/3.1_Easy/largest_element_in_array.cpp
// O(n)

#include<iostream>

using namespace std;

void largestElement(int arr[], int n) {
    int largest = arr[0];
    for (int i = 0; i < n; i++) {
        if (largest < arr[i]) {
            largest = arr[i];
        }
    }
    cout << "The largest number in the given array is " << largest << endl;
}

int main() {
    int arr[] = {3, 2, 1, 5, 2};
    largestElement(arr, sizeof(arr) / sizeof(arr[0]));
    return 0;
}